src/Indigo/Contracts/HomebaseLite/Impl/Admin.hs

```diff
@@ -15,53 +16,86 @@ import Indigo hiding ((*))
 
 import Morley.Util.Interpolate (itu)
 
-setAdmin :: tp :~> Address => IndigoEntrypoint tp
-setAdmin _ = do
+import Indigo.Contracts.HomebaseLite.Types
+import Indigo.Contracts.HomebaseLite.Impl.Utils
+
+checkSenderIsAdmin :: HasStorage Storage => IndigoM ()
+checkSenderIsAdmin =
+  when (sender /= storage #! #sAdmin) do
+    failCustom_ @() #senderIsNotAdmin
+
+setAdmin :: HomebaseLiteEntrypoint Address
+setAdmin newAdmin = do
   description [itu|
     Called by the current `admin` to transfer its role to someone else.
     Takes in input the address of the new `admin` candidate.
-    Note: for security reasons the transfer isn't complete until the new admin candidate calls `accept_admin`.
-    If there is already a candidate, another call to this entrypoint will replace it.
-    If there is already a candidate, calling this with the `admin` address will invalidate the current candidate.
+    Note: for security reasons the transfer isn't complete until the new admin
+    candidate calls `accept_admin`.
+    If there is already a candidate, another call to this entrypoint will
+    replace it.
+    If there is already a candidate, calling this with the `admin` address will
+    invalidate the current candidate.
     |]
-  pass
-
+  checkSenderIsAdmin
+  storage =: storage !! (#sAdminCandidate, newAdmin)
```

Comment:
I didn't check this, but if my memory serves me correctly:
```suggestion
  setStorageField @Storage #sAdminCandidate newAdmin
```
would be slightly more efficient to use.

Have you tried to use it?

Reply:
I don't think it's particularly more efficient, at least I couldn't tell any difference at a (not very close) look. Also, side note, these functions are missing from the reference on https://indigo-lang.gitlab.io/. But anyway, yeah, sorry, I've discovered them, but apparently I've messed up somewhere and those ended up in #11 instead of here :unamused:

Reply:
>  I couldn't tell any difference at a (not very close) look

:thinking:  perhaps the difference disappears after optimization or I might be misremembering too.

> these functions are missing from the reference on https://indigo-lang.gitlab.io/. 

indeed, I'm considering other changes in the storage handling in Indigo, in which case these would be changed or removed. If they won't I'll instead open an issue to add them there.

> I've discovered them, but apparently I've messed up somewhere and those ended up in #11 instead of here

Do you mean before my review or now? 
Because now I see that you've pushed the changes here in fixup commits, so all seems good.

Reply:
> Do you mean before my review or now?

Before your review.